lib/src/main/java/space/taran/arklib/domain/dao/Resource.kt

```diff
@@ -0,0 +1,37 @@
+package space.taran.arklib.domain.dao
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import space.taran.arklib.ResourceId
+import space.taran.arklib.domain.index.ResourceMeta
+import java.nio.file.Path
+
+typealias Milliseconds = Long
+typealias StringPath = String
+
+@Entity
+data class Resource(
+    @PrimaryKey(autoGenerate = false)
+    val id: ResourceId,
+    val root: StringPath,
+    val path: StringPath,
+    val name: String,
+    val extension: String,
+    val modified: Milliseconds,
+    val size: Long,
```

Comment:
We can omit this field now and use `id.dataSize` instead. Tiny shortcut-function can be implemented though like `fun size() = id.dataSize`.